docs/recipes/ActionQueue.md

```diff
@@ -0,0 +1,156 @@
+# Action Queue
+
+## Building a simple action queue
+
+Sometimes it might be desirable to dynamically add commands to a queue and
+handle each one in succession. Lets say for example we want to be able to
+drag tasks on a taskboard. Whenevever a drag movement ends we have to send the
+new position of the task to the server as well, so that it persists on a page reload.
+If our request fails we want to roll back and put the task at its source position.
+But what happens if we drag multiple tasks in quick succession?
+Because of their asynchronous nature requests could resolve in any order.
+Meaning we might produce inconsistent states.
+There would be no way for us to control in which order our requests resolve,
+and because each request is triggered by its own action we cannot make use of
+`Cmd.list(cmds, { sequence: true })`.
+Instead we need to build a queue ourselves which luckily is quite easy to do with redux-loop.
+The example below shows, how you could model your initial state as well as your _QueueReducer_.
+
+```js
+const initialState = {
+  queue: [],
+  error: false,
+  //other fields
+};
+
+function handleItemDraggedAction(state, action){
+  const { queue } = state;
+  const { id } = action.payload;
+
+  if(queue.length){ //something is already running so just add a request to the queue
+    return {
+      ...state,
+      queue: queue.concat([id]) //add new request to back of queue
+    };
+  }
+
+  //nothing is running so just add the new request to the front of the queue and run it
+  const newState = {
+    ...state,
+    queue: [id]
+  };
+
+  return loop(newState, Cmd.run(doRequest, {
+    args: [id],
+    successActionCreator: requestSuccessAction,
+    failActionCreator: requestFailAction
+  }));
+}
+
+function handleRequestSuccessAction(state, action){
+  const newState = {
+    ...state,
+    queue: state.queue.slice(1) //remove old request from front of queue
+  };
+
+  if(!newState.queue.length){
+    //no more requests in the queue
+    return newState;
+  }
+
+  //start the next request
+  return loop(newState, Cmd.run(doRequest, {
+    args: [newState.queue[0]],
+    successActionCreator: requestSuccessAction,
+    failActionCreator: requestFailAction
+  }));
+}
+
+function handleRequestFailedAction(state, action){
+  return {
+    ...state,
+    queue: [], //clear queue on failure
+    error: true
+  }
+}
+```
+
+## Handling different kinds of actions
+
+Sometimes you might want to queue different kinds of actions.
+To achieve this, we could build a `QueueReducer` that listens for specific properties
+instead of an `action.type`:
+
+```js
+function QueueReducer (state, action) {
+  if (action.queuedAction) {
+    // add to queue and potentially dispatch the action with Cmd.action if the
```

Comment:
can you replace these comments with the actual logic? I think it might be easier to understand that way